### PR TITLE
Convert vector elements when needed

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Adds a test case demonstrating the _use of automatic conversions_ when processing vector elements.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: test/Tests/src/Semantic/Conversion_Use.enso [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from Standard.Base import all
+
+type Hello
+    Say (msg:Text)
+
+    formulate : Vector Hello -> Text
```
Comment (JaroslavTulach) [inline test/Tests/src/Semantic/Conversion_Use.enso:6]: First of all there is a static method that operates on a `Vector` with `Hello` elements.

File: test/Tests/src/Semantic/Conversion_Spec.enso [hunk 2]
```diff
@@ -115,4 +116,11 @@ spec =
         Test.specify "should not allow currying" <|
             Panic.recover Any (Foo.from) . catch Any .to_display_text . should_equal "Conversion currying without `that` argument is not supported."
 
+        Test.specify "Use conversions to in Conversion_Use module" <|
+            Hello.formulate [ Hello.Say "Proper", Hello.Say "Type" ] . should_equal "ProperType"
+            Hello.formulate [ Foo.Value "Perform", Bar.Value "Conversion" ] . should_equal "PERFORM conversion!"
```
Comment (JaroslavTulach) [inline test/Tests/src/Semantic/Conversion_Spec.enso:121]: This line invokes the static `formulate` method with non-`Hello` instances, but these instances have conversions methods to `Hello` defined in this module. See below...

File: test/Tests/src/Semantic/Conversion_Use.enso [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    formulate arr =
+        process (t:Text) (h:Hello) = t + h.msg
```
Comment (JaroslavTulach) [inline test/Tests/src/Semantic/Conversion_Use.enso:8]: By using ascribed arguments in a function that `fold`s over the `Vector`, we request each element to be `Hello` or converted to `Hello`.

**Review by JaroslavTulach — COMMENTED** [review]
The fact that the new test case works demonstrates that _logic that locates conversion_ is good. `Hello.formulate` knows nothing about `Foo` or `Bar` types, yet the runtime finds the appropriate conversions that are defined next to definition of `Foo` and/or `Bar`.

**Review by jdunkerley — APPROVED** [review]
Thanks for this test case - this looks awesome. 
Will try it on a couple of Table methods.

**Review by radeusgd — COMMENTED** [review]
🎉

**Pull request merged**